src/calls.ts

```diff
@@ -109,3 +233,186 @@ export function createGetSwapReceipt(config: OnlySwapsConfig, params: GetSwapRec
+    }
+}
+
+/**
+ * Creates a complete Aave V3 supply hook for use in preHooks or postHooks.
+ * This only creates the supply hook - approve hook should be created separately.
+ *
+ * @param params - Parameters for the Aave V3 supply function
+ * @param aaveV3PoolAddress - Target address for Aave V3 Pool contract
+ * @param gasLimit - Optional gas limit for the hook. Defaults to 100000.
+ * @param publicClient - Public client to validate the contract address. Required - validates that the address is a contract and implements the supply function.
+ * @returns Hook object for Aave V3 supply
+ * @throws Error if publicClient is not provided or validation fails
+ */
+export async function createAaveV3SupplyHook(
+    params: AaveV3SupplyParams,
+    aaveV3PoolAddress: Address,
+    gasLimit?: bigint,
+    publicClient?: PublicClient
+): Promise<Hook> {
+    // Validate that publicClient is provided
+    if (!publicClient) {
+        throw new Error("publicClient is required for createAaveV3SupplyHook to validate the Aave V3 contract address")
+    }
+
+    // Validate contract
+    await validateAaveV3Contract(publicClient, aaveV3PoolAddress)
+
+    const hookGasLimit = gasLimit ?? 100_000n
+    return {
+        target: aaveV3PoolAddress,
+        callData: createAaveV3SupplyHookCallData(params),
+        gasLimit: hookGasLimit,
+    }
+}
+
+/**
+ * Creates encoded call data for ERC20 approve function hook.
+ * This can be used directly in preHooks or postHooks when creating swap requests.
+ * 
+ * @param spender - Address to approve for spending tokens
+ * @param amount - Amount of tokens to approve
+ * @returns Encoded function call data as Hex string
+ */
+export function createERC20ApproveHookCallData(spender: Address, amount: bigint): Hex {
+    return encodeFunctionData({
+        abi: erc20Abi,
+        functionName: "approve",
+        args: [spender, amount]
+    })
+}
+
+/**
+ * Creates a complete ERC20 approve hook for use in preHooks or postHooks.
+ * 
+ * @param tokenAddress - Address of the ERC20 token to approve
+ * @param spender - Address to approve for spending tokens
+ * @param amount - Amount of tokens to approve
+ * @param gasLimit - Optional gas limit for the hook execution. Defaults to 100000.
+ * @returns A Hook object ready to use in swap requests
+ * 
+ * @example
+ * ```ts
+ * import { createERC20ApproveHook } from 'onlyswaps-js'
+ * 
+ * // With default gas limit
+ * const approveHook = createERC20ApproveHook(
+ *   USDT_ADDRESS,        // token address
+ *   SPENDER_ADDRESS,     // address to approve
+ *   1000n                // amount to approve
+ * )
+ * 
+ * // With custom gas limit
+ * const approveHookWithCustomGas = createERC20ApproveHook(
+ *   USDT_ADDRESS,
+ *   SPENDER_ADDRESS,
+ *   1000n,
+ *   150_000n             // custom gas limit
+ * )
+ * 
+ * await onlyswaps.swap({
+ *   // ... other params
+ *   preHooks: [approveHook]
+ * })
+ * ```
+ */
+export function createERC20ApproveHook(
```

Comment:
ah interesting, I wasn't sure if we could actually approve funds in a hook. We should probably make it clear this can't happen in a post-hook